packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -206,6 +207,21 @@ export default class {
     const objects = this._xo._objects
 
     forEach(newXapiObjects, function handleObject(xapiObject, xapiId) {
+      const serverIdsByPool = this._serverIdsByPool
+
+      // handle pool UUID change
+      if (
+        xapiObject.$type === 'pool' &&
+        serverIdsByPool[xapiObject.$id] === undefined
```

Comment:
You can use `xapiId` which is the same.

Reply:
We used `pool.$id` as a server entry key. Please see https://github.com/vatesfr/xen-orchestra/pull/4302/files#diff-9b31392ad9c6d4090e97f3073e0d7033R317